src/md5/md5.cc

```diff
@@ -80,7 +80,7 @@ void MD5::update ( const uint1 *input, uint4 input_length) {
   }
 
   // Compute number of bytes mod 64
-  buffer_index = (unsigned int)((count[0] >> 3) & 0x3F);
+  buffer_index = (uint4)((count[0] >> 3) & 0x3F);
```

Comment:
`uint32_t` would be even better. Why change the type at all though?

Reply:
The type usage in this class was a huge mess, i tried the minimum change approach to fix it. ``buffer_index`` is a ``uint4`` so i cast the right side to the left side. Most probably this cast isn't necessary at all because ``count`` is a ``uint4`` as well and the integer literals are so small that they won't introduce a type cast at all, but the cast was there so i kept it.